lib/src/easy_localization_app.dart

```diff
@@ -2,118 +2,103 @@ import 'dart:developer';
-  // _savedLocale is set.
-  _EasyLocalizationLocale(
-      Locale fallbackLocale, List<Locale> supportedLocales, bool saveLocale)
-      : this.saveLocale = saveLocale {
-    _init(fallbackLocale, supportedLocales);
-  }
+class _EasyLocalizationState extends State<EasyLocalization> {
+  Locale locale;
+  EasyLocalizationDelegate delegate;
 
-  //Initialize _EasyLocalizationLocale
-  _init(Locale fallbackLocale, List<Locale> supportedLocales) async {
-    // Get Device Locale
-    _osLocale = await _getDeviceLocale();
-    log('easy localization: Device locale ${_osLocale.toString()}');
-    // If saved locale then get
-    if (_savedLocale != null && this.saveLocale) {
-      log('easy localization: Saved locale loaded ${_savedLocale.toString()}');
-      locale = _savedLocale;
-    } else {
-      locale = supportedLocales.firstWhere((locale) => _checkInitLocale(locale),
-          orElse: () => _getFallbackLocale(supportedLocales, fallbackLocale));
-    }
-    //Set locale
-    if (Intl.defaultLocale == null) locale = _locale;
+  _onLocaleChanged(Locale locale) {
+    setState(() {
+      this.locale = locale;
+    });
   }
 
-  bool _checkInitLocale(Locale locale) {
-    // If suported locale not contain countryCode then check only languageCode
-    if (locale.countryCode == null) {
-      return (locale.languageCode == _osLocale.languageCode);
-    } else {
-      return (locale == _osLocale);
-    }
+  @override
+  void initState() {
+    loadSavedLocale();
+    delegate = EasyLocalizationDelegate(
+        path: widget.path,
+        supportedLocales: widget.supportedLocales,
+        useOnlyLangCode: widget.useOnlyLangCode,
+        assetLoader: widget.assetLoader,
+        onLocaleChanged: _onLocaleChanged);
   }
 
-  //Get fallback Locale
-  Locale _getFallbackLocale(
-      List<Locale> supportedLocales, Locale fallbackLocale) {
-    //If fallbackLocale not set then return first from supportedLocales
-    if (fallbackLocale != null) {
-      return fallbackLocale;
-    } else {
-      return supportedLocales.first;
+  loadSavedLocale() async {
+    SharedPreferences _preferences = await SharedPreferences.getInstance();
+    var _codeLang = _preferences.getString('codeLa');
+    var _codeCoun = _preferences.getString('codeCa');
+    var _strLocale = _preferences.getString('locale');
+    if (_strLocale != null) {
+      log('easy localization: Locale loaded from shared preferences ${Locale(_codeLang, _codeCoun)}');
+      setState(() {
+        locale = _localeFromString(_strLocale);
+      });
     }
+    // TODO reload delegate, set on Material Widget
   }
 
-  // Get Device Locale
-  Future<Locale> _getDeviceLocale() async {
-    final String _deviceLocale = await findSystemLocale();
-    return _localeFromString(_deviceLocale);
+  @override
+  Widget build(BuildContext context) {
+    return EasyLocalizationProvider(
+      widget,
+      this.locale,
```

Comment:
`this.locale`  return device locale when reload app
android output: 
```
[MyApp# locale] null
[log] easy localization: Locale loaded from shared preferences ru_RU
[MyApp# locale] ru_RU
[MyApp# locale] en_US
```

Reply:
@Overman775 plz check this

Reply:
@aissat this is fixed in my last commit.